src/placeos-driver/utilities/transcoder.cr

```diff
@@ -1,13 +1,13 @@
 module PlaceOS::Driver::Utilities::Transcoder
   # Converts a hex encoded string into bytes
-  def hex_to_bytes(string)
+  def hex_to_bytes(string : String) : Bytes
     string = string.gsub(/(0x|[^0-9A-Fa-f])*/, "")
     string = "0#{string}" if string.size % 2 > 0
     string.hexbytes
   end
 
   # Converts a byte array into bytes
-  def array_to_bytes(array)
+  def array_to_bytes(array : Array(Int32 | UInt8)) : Bytes
```

Comment:
Wouldn't this work for all Ints if we don't specify the type?

Reply:
Also thinking it should be `Array(Int32) | Array(UInt8)` if we want to specify a type

Reply:
yeah that's what I wanted to do but the driver test runner gave an error which I took a screenshot of here https://github.com/PlaceOS/driver/pull/32#issue-439973243

Reply:
the screenshot shows `def array_to_bytes(array : Array(Int))`
but what is the issue with `def array_to_bytes(array)`

Reply:
yep sorry @stakach , I should've been more clear but when you add this to any driver `include Utilities::Transcoder`, and you don't specify a type for array, you'll get this error in the driver test runner.

![image](https://user-images.githubusercontent.com/18463731/86580981-58336c80-bf77-11ea-97f9-8fb1d010e508.png)

Looks to be the same error as defining a method and not specifying a type for the parameters in a driver which @KimBurgess said was a requirement for method in drivers which this PR addresses https://github.com/PlaceOS/driver/pull/25

Reply:
Ahh cool, yeah then the solution is to make the methods not public.
we should be leaving them generic and making them `protected`

i.e `protected def array_to_bytes(array)`

that will solve the problem as they are for internal use only

Reply:
Thanks @stakach , I've made that change for both those methods